File: Experimental_Data/subDafny/Leon/subDafny.scala
import leon.lang._

object subDafny {

 def subDafny(x : BigInt, y : BigInt): BigInt = ({
 require(x >= 0 && y>=0)
     var r = x
     if(y < 0) {
       var n = y
       while(n != 0) {
         r = r + 1
         n = n + 1
       } 
     } else {
       var n = y
       while(n != 0) {
         r = r - 1
         n = n - 1
       }
     }
     r
  }) ensuring(res => res == x-y)

}

File: Experimental_Data/subDafny/subDafny.java

public class subDafny{

	
	public void subDafny(int X,int Y) {
	  
        int r;
	int l;
	r = X;
    if(Y < 0) 
    {
       l = Y;
       while(l != 0) 
      {
        	r = r + 1;
        	l = l + 1;
      }
    } 
   else 
    {
      l = Y;
      while(l != 0) {
        r = r - 1;
        l = l - 1;
      }
    }	
	
 }
}
